src/CoreGraphics/CGPDFArray.cs

```diff
@@ -142,5 +142,41 @@ public bool GetString (int idx, out string result)
 			result = r ? CGPDFString.ToString (res) : null;
 			return r;
 		}
+
+		delegate bool ApplyBlockHandlerDelegate (IntPtr block, nint index, IntPtr value, IntPtr info);
+		static readonly ApplyBlockHandlerDelegate applyblock_handler = ApplyBlockHandler;
+
+		[MonoPInvokeCallback (typeof (ApplyBlockHandlerDelegate))]
+		static unsafe bool ApplyBlockHandler (IntPtr block, nint index, IntPtr value, IntPtr info)
+		{
+			var del = (ApplyBlockCallback) ((BlockLiteral *) block)->Target;
+			if (del != null)
+				return del (index, new CGPDFObject (value), info);
+
+			return false;
+		}
+
+		public delegate bool ApplyBlockCallback (nint index, CGPDFObject value, IntPtr info);
+
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		[iOS (12, 0)][Mac (10, 14)][TV (12, 0)][Watch (5, 0)]
+		extern static bool CGPDFArrayApplyBlock (/* CGPDFArrayRef */ IntPtr array, /* CGPDFArrayApplierBlock */ ref BlockLiteral block, /* void* */ IntPtr info);
+
+		[iOS (12, 0)][Mac (10, 14)][TV (12, 0)][Watch (5, 0)]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public bool ApplyBlock (ApplyBlockCallback callback, IntPtr info = default (IntPtr))
```

Comment:
having an `IntPtr` makes it harder to pass state to the callback, than an `System.Object`

Reply:
That's a good point, we could wrap an `object` with a GCHandle.